### PR TITLE
Various API refactorings

### PR DESCRIPTION
**Pull request by davidmehren**
### Component/Part
API

### Description
This PR contains a bunch of changes to the API. See the commit messages for details.

### Steps

- [x] Added implementation
- [x] Added / updated tests
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### CODE REVIEW

**Comment by davidmehren** [comment]
This is best reviewed commit-by-commit.

File: test/private-api/auth.e2e-spec.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ describe('Auth', () => {
   describe('POST /auth/local', () => {
     it('works', async () => {
       const registrationDto: RegisterDto = {
-        displayname: displayname,
+        displayName: displayname,
```
Comment (InnayTool) [inline test/private-api/auth.e2e-spec.ts:44]:
```suggestion
        displayName: displayName,
```

File: test/private-api/auth.e2e-spec.ts [hunk 2]
```diff
@@ -67,7 +67,7 @@ describe('Auth', () => {
         const username2 = 'already_existing';
         await testSetup.userService.createUser(username2, displayname);
         const registrationDto: RegisterDto = {
-          displayname: displayname,
+          displayName: displayname,
```
Comment (InnayTool) [inline test/private-api/auth.e2e-spec.ts:70]:
```suggestion
          displayName: displayName,
```

File: test/private-api/auth.e2e-spec.ts [hunk 3]
```diff
@@ -80,7 +80,7 @@ describe('Auth', () => {
       it('when registration is disabled', async () => {
         testSetup.configService.get('authConfig').local.enableRegister = false;
         const registrationDto: RegisterDto = {
-          displayname: displayname,
+          displayName: displayname,
```
Comment (InnayTool) [inline test/private-api/auth.e2e-spec.ts:83]:
```suggestion
          displayName: displayName,
```

File: src/revisions/revisions.service.ts [hunk 4]
```diff
@@ -101,20 +104,61 @@ export class RevisionsService {
...
+  async getRevisionAuthors(revision: Revision): Promise<Author[]> {
+    const authors = [];
+    for (const edit of await revision.edits) {
+      authors.push(await edit.author);
+    }
+    return authors;
+  }
+
+  async getRevisionAuthorUsernames(revision: Revision): Promise<string[]> {
+    const authors = await this.getRevisionAuthors(revision);
```
Comment (InnayTool) [inline src/revisions/revisions.service.ts:116]: I think we should change the implementation of this function to the implementation of the `getRevisionAuthors` function.
The current implementation is more complicated and runs the loop twice.

**Review by InnayTool — CHANGES_REQUESTED** [review]

File: src/notes/note.entity.ts [hunk 5]
```diff
@@ -89,6 +90,14 @@ export class Note {
   @JoinTable()
   tags: Promise<Tag[]>;
 
+  @Column({
+    default: 2,
```
Comment (DerMolly) [inline src/notes/note.entity.ts:94]: I think this redundant

**Review by ErikMichelson — COMMENTED** [review]
https://github.com/hedgedoc/hedgedoc/pull/2059/files#diff-c8b489e9eaa391c14ed3bb54a048dd06770931ad0ba07904bdca2bfc44c5e2f5R234

`toMediaUploadUrlDto` seem unused

**Review by ErikMichelson — COMMENTED** [review]
`HistoryEntryImportDto` still contains `lastVisited` instead of `lastVisitedAt`

File: src/api/private/me/me.controller.ts [hunk 6]
```diff
@@ -56,10 +61,19 @@ export class MeController {
   }
 
   @Post('profile')
```
[the remark this answers is not in the document]
Comment (davidmehren) [reply src/api/private/me/me.controller.ts:63]: IDK, but let's ignore this issue in this PR.

[the remark this answers is not in the document]
Comment (davidmehren) [reply src/api/private/me/me.controller.ts:63]: IDK, but let's ignore this issue in this PR.

File: src/revisions/revisions.service.ts (hunk 4 above)
Comment (InnayTool) [reply src/revisions/revisions.service.ts:113]:
```
  async getRevisionAuthorUsernames(revision: Revision): Promise<string[]> {
    const usernames: string[] = [];
    for (const edit of await revision.edits) {
      const user = await (await edit.author).user;
      if (user) {
        usernames.push(user.username);
      }
    }
    return usernames;
  }
```

File: src/notes/note.entity.ts (hunk 5 above)
Comment (davidmehren) [reply src/notes/note.entity.ts]: It is, but this statement cannot be removed, otherwise the app crashes on startup with a constraint violation.

**Review by InnayTool — APPROVED** [review]

**Review by DerMolly — APPROVED** [review]

**Pull request merged**